### PR TITLE
New Resource: `azurerm_data_protection_backup_instance_postgresql_flexible_server`

### PR DESCRIPTION
**Pull request by neil-yechenwei**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description
This PR is to support Data Protection Backup Instance for Postgresql FS.

API Reference: https://github.com/Azure/azure-rest-api-specs/tree/main/specification/dataprotection/resource-manager/Microsoft.DataProtection/stable/2024-04-01
Overview: https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-backup-restore#long-term-retention-preview
This PR is referring the existing Backup Instances in RP https://github.com/hashicorp/terraform-provider-azurerm/tree/main/internal/services/dataprotection.

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [x] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

```
--- PASS: TestAccDataProtectionBackupInstancePostgreSQLFlexibleServer_requiresImport (1406.99s)
--- PASS: TestAccDataProtectionBackupInstancePostgreSQLFlexibleServer_basic (730.69s)
--- PASS: TestAccDataProtectionBackupInstancePostgreSQLFlexibleServer_update (1274.70s)
```

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* New Resource: `azurerm_data_protection_backup_instance_postgresql_flexible_server`

This is a (please select all that apply):

- [ ] Bug Fix
- [x] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes https://github.com/hashicorp/terraform-provider-azurerm/issues/24036

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: internal/services/dataprotection/data_protection_backup_instance_postgresql_flexible_server_resource.go [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+			vaultId := backupinstances.NewBackupVaultID(id.SubscriptionId, id.ResourceGroupName, id.BackupVaultName)
+
+			state := BackupInstancePostgreSQLFlexibleServerModel{
+				Name:    id.BackupInstanceName,
+				VaultId: vaultId.ID(),
+			}
+
+			if model := resp.Model; model != nil {
+				if props := model.Properties; props != nil {
+					state.Location = location.Normalize(pointer.From(props.DataSourceInfo.ResourceLocation))
```
Comment (ms-zhenhua) [inline internal/services/dataprotection/data_protection_backup_instance_postgresql_flexible_server_resource.go:194]:
```suggestion
					state.Location = location.NormalizeNilable(props.DataSourceInfo.ResourceLocation)
```

**Review by ms-zhenhua — CHANGES_REQUESTED** [review]
Hi @neil-yechenwei ,

Thanks for this PR - I've taken a look through and left some comments inline. If we can fix those up, this should be good to go 👍

File: internal/services/dataprotection/data_protection_backup_instance_postgresql_flexible_server_resource.go [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+	return backupinstances.ValidateBackupInstanceID
+}
+
+func (r DataProtectionBackupInstancePostgreSQLFlexibleServerResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
```
[the remark this answers is not in the document]
Comment (neil-yechenwei) [reply internal/services/dataprotection/data_protection_backup_instance_postgresql_flexible_server_resource.go:53]: Sorry. I didn't find name validation from anywhere.

File: internal/services/dataprotection/data_protection_backup_instance_postgresql_flexible_server_resource.go (hunk 1 above)
Comment (neil-yechenwei) [reply internal/services/dataprotection/data_protection_backup_instance_postgresql_flexible_server_resource.go]: Updated

**Comment by neil-yechenwei** [comment]
@ms-zhenhua , thanks for the comments. I updated PR. Please take another look.

**Review by ms-zhenhua — APPROVED** [review]
LGTM~

**Review by katbyte — APPROVED** [review]
LGTM 🚜

**Pull request merged**